File: backup_20250923_180339/frontend/src/types/medical/vaccination.ts
// Vaccination Types for PawfectPal

export interface Vaccination {
  id?: number;
  petId: number;
  vaccineName: string;
  dateAdministered: string; // ISO date string
  nextDueDate?: string; // ISO date string
  batchNumber?: string;
  manufacturer?: string;
  veterinarian: string;
  clinic: string;
  doseNumber?: number;
  notes?: string;
  isCompleted: boolean;
  reminderSent: boolean;
  createdAt?: string; // ISO datetime string
  updatedAt?: string; // ISO datetime string
}

export interface VaccinationCreate {
  petId: number;
  vaccineName: string;
  dateAdministered: string;
  nextDueDate?: string;
  batchNumber?: string;
  manufacturer?: string;
  veterinarian: string;
  clinic: string;
  doseNumber?: number;
  notes?: string;
  isCompleted?: boolean;
  reminderSent?: boolean;
}

export interface VaccinationUpdate {
  vaccineName: string;
  dateAdministered: string;
  nextDueDate?: string;
  batchNumber?: string;
  manufacturer?: string;
  veterinarian: string;
  clinic: string;
  doseNumber?: number;
  notes?: string;
  isCompleted: boolean;
  reminderSent: boolean;
}

export interface VaccinationListResponse {
  vaccinations: Vaccination[];
  total: number;
  page: number;
  pageSize: number;
}

export interface VaccinationSummary {
  petId: number;
  totalVaccinations: number;
  upToDate: boolean;
  nextDueDate?: string; // ISO date string
  overdueCount: number;
  completedSeries: string[];
}

export interface VaccinationReminder {
  vaccinationId: number;
  petId: number;
  petName: string;
  vaccineName: string;
  dueDate: string; // ISO date string
  daysUntilDue: number;
  isOverdue: boolean;
}

export interface VaccinationFilters {
  vaccineName?: string;
  dateFrom?: string;
  dateTo?: string;
  veterinarian?: string;
  clinic?: string;
  isCompleted?: boolean;
  isOverdue?: boolean;
  page?: number;
  pageSize?: number;
}

// Common vaccine schedules
export interface VaccineScheduleItem {
  name: string;
  ageWeeks: number[];
  boosterInterval?: number; // months
  required: boolean;
  description?: string;
}

export interface VaccineSchedule {
  petType: 'dog' | 'cat' | 'other';
  vaccines: VaccineScheduleItem[];
}

// Standard vaccination schedules
export const DOG_VACCINE_SCHEDULE: VaccineSchedule = {
  petType: 'dog',
  vaccines: [
    {
      name: 'DHPP (Distemper, Hepatitis, Parvovirus, Parainfluenza)',
      ageWeeks: [6, 9, 12, 16],
      boosterInterval: 12,
      required: true,
      description: 'Core vaccine series for puppies'
    },
    {
      name: 'Rabies',
      ageWeeks: [12],
      boosterInterval: 12,
      required: true,
      description: 'Required by law in most areas'
    },
    {
      name: 'Bordetella',
      ageWeeks: [8, 12],
      boosterInterval: 12,
      required: false,
      description: 'Kennel cough prevention'
    }
  ]
};

export const CAT_VACCINE_SCHEDULE: VaccineSchedule = {
  petType: 'cat',
  vaccines: [
    {
      name: 'FVRCP (Feline Viral Rhinotracheitis, Calicivirus, Panleukopenia)',
      ageWeeks: [6, 9, 12, 16],
      boosterInterval: 12,
      required: true,
      description: 'Core vaccine series for kittens'
    },
    {
      name: 'Rabies',
      ageWeeks: [12],
      boosterInterval: 12,
      required: true,
      description: 'Required by law in most areas'
    },
    {
      name: 'FeLV (Feline Leukemia)',
      ageWeeks: [8, 12],
      boosterInterval: 12,
      required: false,
      description: 'For outdoor cats'
    }
  ]
};

File: backend/app/schemas/vaccination.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import date, datetime

class VaccinationBase(BaseModel):
    vaccine_name: str = Field(..., min_length=1, max_length=255)
    vaccine_type: Optional[str] = Field(default="Core", max_length=100)
    date_administered: date
    next_due_date: Optional[date] = None
    batch_number: Optional[str] = Field(None, max_length=100)
    manufacturer: Optional[str] = Field(None, max_length=255)
    veterinarian: str = Field(..., min_length=1, max_length=255)
    clinic: str = Field(..., min_length=1, max_length=255)
    dose_number: Optional[int] = Field(default=1, ge=1)
    notes: Optional[str] = None
    is_completed: bool = Field(default=True)
    reminder_sent: bool = Field(default=False)

class VaccinationCreate(VaccinationBase):
    pet_id: int

class VaccinationUpdate(VaccinationBase):
    pass

class VaccinationResponse(VaccinationBase):
    id: int
    pet_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

class VaccinationSummary(BaseModel):
    pet_id: int
    total_vaccinations: int
    up_to_date: bool
    next_due_date: Optional[date] = None
    overdue_count: int
    due_soon_count: int
    completed_series: List[str]

class VaccinationListResponse(BaseModel):
    vaccinations: List[VaccinationResponse]
    total: int
    page: int
    page_size: int

class VaccinationReminder(BaseModel):
    vaccination_id: int
    pet_id: int
    pet_name: str
    vaccine_name: str
    due_date: date
    days_until_due: int
    is_overdue: bool